packages/notebook-extension/src/index.tsx

```diff
@@ -1060,6 +1065,87 @@ const activeCellTool: JupyterFrontEndPlugin<void> = {
   }
 };
 
+export const filterCellsPlugin: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/notebook-extension:filter-cells',
+  description: 'Filter cells.',
+  autoStart: true,
+  requires: [INotebookTracker, IFormRendererRegistry],
+  optional: [ICommandPalette, ITranslator],
+  activate: (
+    app: JupyterFrontEnd,
+    tracker: INotebookTracker,
+    formRegistry: IFormRendererRegistry,
+    palette: ICommandPalette | null,
+    translator_: ITranslator | null
+  ): void => {
+    const translator = translator_ ?? nullTranslator;
+    const trans = translator.load('jupyterlab');
+    const { shell, commands } = app;
+    const isEnabled = () => Private.isEnabled(shell, tracker);
+    const notebookModelMap = new WeakMap<Notebook, CellFiltersModel>();
+
+    const filterButton: IFormRenderer = {
+      fieldRenderer: () => (
+        <CommandToolbarButtonComponent
+          commands={commands}
+          id={CommandIDs.filterCells}
+          caption={trans.__('Open the filtering tool')}
+        ></CommandToolbarButtonComponent>
+      )
+    };
+    formRegistry.addRenderer(
+      '@jupyterlab/notebook-extension:filter-cells.renderer',
+      filterButton
+    );
+
+    app.commands.addCommand(CommandIDs.filterCells, {
+      label: trans.__('Filter Cells'),
+      caption: trans.__('Filter cells'),
+      execute: async args => {
+        const current = getCurrent(tracker, shell, args);
+
+        if (current) {
+          let model = notebookModelMap.get(current.content);
+          if (!model) {
+            model = new CellFiltersModel({ notebook: current.content });
+            notebookModelMap.set(current.content, model);
+          }
+
+          const selectButton = Dialog.okButton({
+            label: trans.__('Filter Cell(s)'),
+            actions: ['select']
+          });
+          const clearButton = Dialog.warnButton({
+            label: trans.__('Clear Filter(s)'),
```

Comment:
How hard is it to know if plural/singular should be used? This would help with translations because not every language can pluralise by adding a short suffix.

Reply:
@HaudinFlorence I have add for testing a counter for the number of active filters in the `CellFiltersComponent`. I guess that then using this counter it would be possible to set something like what is made for the commands linked to the buttons of notebook toolbars something similar to:
```ts
commands.addCommand(CommandIDs.runAndAdvance, {
    label: trans.__('Run Selected Cells'),
    caption: trans.__('Run the selected cells'),
    label: args => {
      const current = getCurrent(tracker, shell, args);
      return trans._n(
        'Run Selected Cell',
        'Run Selected Cells',
        current?.content.selectedCells.length ?? 1
      );
    },
    caption: args => {
      const current = getCurrent(tracker, shell, args);
      return trans._n(
        'Run this cell and advance',
        'Run these %1 cells and advance',
        current?.content.selectedCells.length ?? 1
      );
    },
    ```